homeassistant/components/smappee_official/switch.py

```diff
@@ -0,0 +1,198 @@
+            )
+
+        # Switch or comfort plug
+        return self._state == "ON_ON"
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return ICON
+
+    def turn_on(self, **kwargs):
+        """Turn on Comport Plug."""
+        if self._actuator_type in ["SWITCH", "COMFORT_PLUG"]:
+            self._service_location.set_actuator_state(self._actuator_id, state="ON_ON")
+        elif self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            self._service_location.set_actuator_state(
+                self._actuator_id, state=self._actuator_state_option
+            )
+        self._state = self._service_location.actuators.get(self._actuator_id).state
+
+    def turn_off(self, **kwargs):
+        """Turn off Comport Plug."""
+        if self._actuator_type in ["SWITCH", "COMFORT_PLUG"]:
+            self._service_location.set_actuator_state(
+                self._actuator_id, state="OFF_OFF"
+            )
+        elif self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            self._service_location.set_actuator_state(
+                self._actuator_id, state="PLACEHOLDER", api=False
+            )
+        self._state = self._service_location.actuators.get(self._actuator_id).state
+
+    @property
+    def available(self):
+        """Return True if entity is available. Unavailable for COMFORT_PLUGS."""
+        return (
+            self._connection_state == "CONNECTED"
+            or self._actuator_type == "COMFORT_PLUG"
+        )
+
+    @property
+    def today_energy_kwh(self):
+        """Return the today total energy usage in kWh."""
+        if self._actuator_type == "SWITCH":
+            cons = self._service_location.actuators.get(
+                self._actuator_id
+            ).consumption_today
+            if cons is not None:
+                return round(cons / 1000.0, 2)
+        return None
+
+    @property
+    def unique_id(self,):
+        """Return the unique ID for this switch."""
+        if self._actuator_type == "INFINITY_OUTPUT_MODULE":
+            return (
+                f"{self._service_location.device_serial_number}-"
+                f"{self._service_location.service_location_id}-actuator-"
+                f"{self._actuator_id}-{self._actuator_state_option}"
+            )
+
+        # Switch or comfort plug
+        return (
+            f"{self._service_location.device_serial_number}-"
+            f"{self._service_location.service_location_id}-actuator-"
+            f"{self._actuator_id}"
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return {
+            "Service location name": self._service_location.service_location_name,
+            "Device serialnumber": self._service_location.device_serial_number,
+            "Actuator name": self._actuator_name,
+            "Actuator type": self._actuator_type,
+            "Today energy kwh": self.today_energy_kwh,
```

Comment:
The today_energy_kwh attribute will be included by the base switch entity class. Please remove it here.